dotCMS/src/main/java/com/dotmarketing/portlets/categories/business/CategoryFactoryImpl.java

```diff
@@ -843,4 +835,40 @@ protected String suggestVelocityVarName(final String categoryVelVarName) throws
         throw new DotDataException("Unable to suggest a variable name.  Got to:" + var);
     }
 
+	/**
+	 *  Default Implementation for {@link CategoryFactory#findAll(String)}
+	 * @param filter Value used to filter the Category by, returning only Categories that contain this value in their key, name, or variable name
+	 *
+	 * @return
+	 */
+	public List<Category> findAll(final String filter) throws DotDataException {

```

Comment:
Missing DB annotation

Reply:
done in the API PR dotCMS/src/main/java/com/dotmarketing/portlets/categories/business/CategoryAPIImpl.java